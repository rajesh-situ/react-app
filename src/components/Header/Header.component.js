import React, { Component } from 'react';
import PropTypes from 'prop-types';
import * as styles from './Header.styles';

export default class Header extends Component {
  render() {
    const { headerText } = this.props;
    return (
      <styles.Container>
        <styles.Title>
          {headerText}
        </styles.Title>
      </styles.Container>
    );
  }
}

Header.propTypes = {
  headerText: PropTypes.string
};

Header.defaultProps = {
  headerText: ''
};
