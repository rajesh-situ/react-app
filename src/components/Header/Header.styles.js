import styled from 'styled-components';

export const Container = styled.div`
    border-top: 5px solid var(--main-header-color);
    box-shadow: 0 0 15px 0 var(--main-shadow-color);
    display: flex;
    justify-content: space-between;
    align-items: center;
    flex-wrap: nowrap;
    height: 65px;
    padding: 0 40px;
`;

export const Title = styled.h1`
    font-size: var(--font-size-medium);
`;