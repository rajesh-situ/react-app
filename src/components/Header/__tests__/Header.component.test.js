import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Header.component';

describe('The header component', () => {
  it('should render correctly', () => {
    const component = renderer.create(<Header />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
