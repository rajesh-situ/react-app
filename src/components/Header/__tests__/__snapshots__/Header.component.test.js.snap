// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The header component should render correctly 1`] = `
<div
  className="Headerstyles__Container-sc-13gsfyb-0 gNvquE"
>
  <h1
    className="Headerstyles__Title-sc-13gsfyb-1 kFhJju"
  >
    
  </h1>
</div>
`;
